src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -37,6 +38,40 @@ public T WithHttpClientFactory(IMsalHttpClientFactory httpClientFactory)
             return (T)this;
         }
 
+#pragma warning disable CS1570 // XML comment has badly formed XML
+        /// <summary>
+        /// Allows developers to configure details about the authorities used, bypassing the need for MSAL
+        /// to perform instance discovery, e.g. https://login.microsoftonline.com/common/discovery/instance?api-version=1.1&authorization_endpoint=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fv2.0%2Fauthorize
+        /// MSAL uses the aliases to identify tokens that were issued to equivalent authorities, as tokens issued for one authority in the alias set is 
+        /// valid for all other aliases. For more details see https://aka.ms/msal-net-custom-instance-metadata
+        /// </summary>
+        /// <remarks>
+        /// For security reasons, do not use this option if the authority is not a constant. 
```

Comment:
what does it mean for the authority to be constant?

I would write something like:
"Don't use this method unless you have specific needs that MSAL.NET's default behavior (fetching the instance metadata from the Azure AD discovery endpoint) does not fullfil"

Reply:
The reason this whole validate authority was introdcued was because in the old days you could send an unauthenticated request to a priotected API and it would reply with an Unauthorized with a header "Authority=www.someauthority.com". Developers take the authority and pass it on to ADAL / MSAL. 

A hacked API could give send you to a hacked authority.

Reply:
Will update the comment though.